src-russian/interlude_projecting_children.md

```diff
@@ -0,0 +1,175 @@
+# Пробрасывание дочерних элементов
+
+В процессе построения компонентов, может возникнуть желание как бы "пробросить" дочерний элемент сквозь несколько слоёв компонентов.
+
+## Суть проблемы
+
+Рассмотрим следующее:
+
+```rust
+pub fn LoggedIn<F, IV>(fallback: F, children: ChildrenFn) -> impl IntoView
+where
+    F: Fn() -> IV + 'static,
+    IV: IntoView,
+{
+    view! {
+        <Suspense
+            fallback=|| ()
+        >
+            <Show
+				// check whether user is verified
+				// by reading from the resource
+                when=move || todo!()
+                fallback=fallback
+            >
+				{children()}
+			</Show>
+        </Suspense>
+    }
+}
+```
+
+Все довольно прозаично: когда пользователь "вошёл", показываем `children`. Если же нет, то отображаем `fallback`. А пока ожидаем ответа, рендерим `()`, т.е. ничего.
+
+Другими словами, пытаемся передать дочерний элемент `<LoggedIn/>` _сквозь_ `<Suspense/>`, чтобы тот уже стал дочерним компонента `<Show/>`. Что-то подобное можно понимать под "пробрасыванием".
+
+Только дело в том, что этот код не скомпелируется:
+
+```
+error[E0507]: cannot move out of `fallback`, a captured variable in an `Fn` closure
+error[E0507]: cannot move out of `children`, a captured variable in an `Fn` closure
+```
+
+Проблема в том, что и `<Suspense/>`, и `<Show/>` должны иметь возможность конструировать своих `children` несколько раз. При первом построении дочерних элементов `<Suspense/>`, им будет взято право владения и `fallback`, и `children`, чтобы можно было передать их для вызова в `<Show/>`. Однако, из-за этого они перестают быть доступны для построения будущих дочерних компонентов `<Suspense/>`.
```

Comment:
Должны? Конструировать?

Reply:
> The first time you construct `<Suspense/>`’s children, it would take ownership of `fallback` and `children` to move them into the invocation of `<Show/>`, but then they're not available for future `<Suspense/>` children construction.

ты переводишь как

> При первом построении дочерних элементов `<Suspense/>`, им будет взято право владения и `fallback`, и `children`, чтобы можно было передать их для вызова в `<Show/>`. Однако, из-за этого они перестают быть доступны для построения будущих дочерних компонентов `<Suspense/>`.

что по смыслу неправильно. 

> При первом составлении дочерних элементов `<Suspense>` потребовалось бы владение `fallback` и `children`, чтобы сделать `move` этих значений внутрь вызова `<Show/>`, но тогда они не были бы доступны для последующих составлений дочерних элементов `<Suspense/>.

```

Reply:
Здесь мне кажется больше вкусовщина в отношении "построить" и "составить". Однако в остальном твой вариант действительно легче передаёт суть

Reply:
Ты правда не понимаешь разницу в смысловой нагрузке?

Reply:
Если она и есть, я думаю "построить" здесь уместнее. Компоненты строятся билдером